sync/sync_test.go

```diff
@@ -323,32 +322,30 @@ func TestSync_InvalidSyncTarget(t *testing.T) {
 	// generate 300 more headers
 	headers := suite.GenDummyHeaders(300)
 	// malform the remote store's head so that it can serve
-	// the syncer a "bad" sync target that passes initial validation,
-	// but not verification.
+	// the syncer a soft failed head
 	maliciousHeader := headers[299]
 	maliciousHeader.VerifyFailure = true
+	maliciousHeader.SoftFailure = true
 	err = remoteStore.Append(ctx, headers...)
 	require.NoError(t, err)
 
 	// start syncer so that it immediately requests the bad
 	// sync target from the remote peer via Head request
 	err = syncer.Start(ctx)
 	require.NoError(t, err)
-
```

Comment:
Nit: IMO this newline was in a good place.

Reply:
likely by accident, can revert

Reply:
Up to you.